packages/main/src/plugin/configuration-registry.spec.ts

```diff
@@ -28,11 +26,34 @@ import { ConfigurationRegistry } from './configuration-registry.js';
 import type { Directories } from './directories.js';
 import type { NotificationRegistry } from './tasks/notification-registry.js';
 
+// mock the fs module
+vi.mock('node:fs', () => ({
+  readFileSync: vi.fn(),
+  writeFileSync: vi.fn(),
+  cpSync: vi.fn(),
+  promises: {
+    access: vi.fn(),
+    mkdir: vi.fn(),
+    writeFile: vi.fn(),
+    readFile: vi.fn(),
+    copyFile: vi.fn(),
+  },
+}));
+
+// Import fs after mocking
+const fs = await import('node:fs');
+
 let configurationRegistry: ConfigurationRegistry;
 
-// mock the fs methods
-const readFileSync = vi.spyOn(fs, 'readFileSync');
-const cpSync = vi.spyOn(fs, 'cpSync');
+// Access mocked functions
+const readFileSync = vi.mocked(fs.readFileSync);
+const writeFileSync = vi.mocked(fs.writeFileSync);
+const cpSync = vi.mocked(fs.cpSync);
+const accessMock = vi.mocked(fs.promises.access);
+const mkdirMock = vi.mocked(fs.promises.mkdir);
+const writeFileMock = vi.mocked(fs.promises.writeFile);
+const readFileMock = vi.mocked(fs.promises.readFile);
+const copyFileMock = vi.mocked(fs.promises.copyFile);
```

Comment:
just a comment, but I am not sure to understand why you would prefer declaring them at the top of the file, when you could just use v.mocked directly when needed